crates/task-impls/src/quorum_vote/mod.rs

```diff
@@ -328,6 +300,28 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static, V: Versions> Handl
                 _ => {}
             }
         }
+
+        let Some(parent_view_number) = parent_view_number else {
+            tracing::error!(
+                "We don't have the parent_view_number for this view {:?}, but we should, because the vote dependencies have completed.",
+                self.view_number
+            );
+            return;
+        };
+
+        let validated_view = self
```

Comment:
So part of the reason we waited for this view before was because the proposal might still be being processed or fetched.  The reason it all still works ok is because we are probably never behind in tests, and are quick to fetch proposals from others.  I do think there will be a case where returning early here will result in us not voting in a view we otherwise would have.

Further in `update_shared_state` we'll actually call `fetch_proposal` if we don't have the validated state for the view.  Now that code path can't get hit.